build/tasks/RetryCommand.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Diagnostics;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace RepoTasks
+{
+    public class RetryCommand : Task
```

Comment:
We already have a [task](https://github.com/aspnet/BuildTools/blob/dev/modules/BuildTools.Tasks/Run.cs) for this in KoreBuild. IIRC the syntax will be
```
<Run FileName="docker" Command="run ...." MaxRetries="1" />
```

I suggest you use that since Process.Start is actually not going to work well with MSBuild logging.